.github/workflows/main.yml

```diff
@@ -1,27 +1,22 @@
 name: Main workflow
+
 on: [push]
+
 jobs:
   checks:
+    timeout-minutes: 20
+
     env:
       NODE_ENV: test
       DATABASE_URL: postgres://
       AUTH_SECRET: test
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        node: ["16.x"]
-        os: [ubuntu-latest]
-    steps:
-      - name: Checkout repo
-        uses: actions/checkout@v2
 
-      - name: Setup Node.js ${{ matrix.node }}
-        uses: actions/setup-node@v2
-        with:
-          node-version: ${{ matrix.node }}
-          cache: "yarn"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/ci-setup
```

Comment:
Even though I was able to move setup+install to a single `ci-setup` action, we still need to have checkout as a separate step. This can't be put in the `ci-setup` because `ci-setup` is local, it's in this repo - and the CI runner starts without any repo files downloaded. So we first need to fetch them (with `actions/checkout`) and only then we can start using our local `ci-setup` action